apps/test/unit/lib/kits/maker/boards/microBit/LightSensorTest.js

```diff
@@ -60,11 +64,48 @@ describe('LightSensor', function() {
       let stopSpy = sinon.spy(boardClient, 'stopStreamingAnalogChannel');
       lightSensor.start();
       expect(startSpy).to.have.been.calledOnce;
-      expect(startSpy).to.have.been.calledWith(sensor_channels.lightSensor);
+      expect(startSpy).to.have.been.calledWith(SENSOR_CHANNELS.lightSensor);
 
       lightSensor.stop();
       expect(stopSpy).to.have.been.calledOnce;
-      expect(stopSpy).to.have.been.calledWith(sensor_channels.lightSensor);
+      expect(stopSpy).to.have.been.calledWith(SENSOR_CHANNELS.lightSensor);
+    });
+  });
+
+  describe(`getAveragedValue`, () => {
+    it(`returns average of values in buffer`, () => {
+      lightSensor.state.buffer = new Float32Array(
+        MAX_SENSOR_BUFFER_DURATION / SAMPLE_INTERVAL
+      );
+      for (let i = 0; i < lightSensor.state.buffer.length; i++) {
+        lightSensor.state.buffer[i] = i;
+      }
+
+      lightSensor.state.currentBufferWriteIndex = 20;
+      expect(lightSensor.getAveragedValue(1000)).to.equal(9.5);
+
+      lightSensor.state.currentBufferWriteIndex = 3001;
+      expect(lightSensor.getAveragedValue(3000)).to.equal(29.5);
+
+      lightSensor.state.currentBufferWriteIndex = 1;
+      expect(lightSensor.getAveragedValue(50)).to.equal(0);
+
+      lightSensor.state.currentBufferWriteIndex = 20;
+      expect(lightSensor.getAveragedValue(3000)).to.equal(9.5);
+    });
+
+    it(`returns average of values in buffer within the set range`, () => {
+      lightSensor.setRange(10, 110);
+
+      lightSensor.state.buffer = new Float32Array(
+        MAX_SENSOR_BUFFER_DURATION / SAMPLE_INTERVAL
+      );
+      for (let i = 0; i < lightSensor.state.buffer.length; i++) {
+        lightSensor.state.buffer[i] = 5;
+      }
+
+      lightSensor.state.currentBufferWriteIndex = 20;
+      expect(lightSensor.getAveragedValue(800)).to.equal(11.96);
     });
   });
```

Comment:
Yup! Covered in the 4th expect statement in "returns average of values in buffer".

Reply:
Ah nice.  These cases are interesting enough that a little comment above each might be of interest.  Possibly we could also explain why 3000 is special, or just use the existing constant for it instead.

I actually ran the code manually to try some different cases and displayed each index that was being summed, because this sliding window stuff is tricky.  Definitely a case where upfront unit tests can help to validate everything is working as expected!

Reply:
Thanks for updating the test comments.  This coverage looks to be good.

I would consider these tests to be "integration tests" where we have added some coverage to prevent future regressions.  

As mentioned, I manually ran this code manually and hand-inspected every index that they consumed to make sure there were no off-by-one or other odd edge cases, which is very common for this type of code.  As a philosophical note, I'd recommend that, where possible, we make sure that unit testing for this kind of algorithm is set up so that we can very deliberately understand what the code is doing and we can easily hand-verify the result value that comes out.  Where we have a value like `11.96` as a result here, it will hopefully serve as protection against a future regression, but we are assuming that `11.96` is the appropriate result in the first place, and presumably got it as output from a previous run of the code, which is why I consider it more of an integration test than a pure upfront unit test, if that makes sense.